### PR TITLE
fix: Asset module code for better readability

### PR DESCRIPTION
**Pull request by khushi8112**
# Asset Module Refactor PR

This PR refactors the asset module to improve code structure, fix bugs, and enhance maintainability. Testing is a priority to ensure no regressions are introduced.

### Asset Repair
- [x] Asset Repair Depreciation
- [x] Asset Repair (UI)
- [x] Test cases

### Asset Value Adjustment
- [x] Refactored and improved logic
- [x] Test cases

### Depreciation Methods
- [x] Shift-Based Depreciation
- [x] Straight-Line Method
- [x] Daily Pro-Rata Depreciation
- [x] Written Down Value Method
- [x] Journal Entry for Depreciation
- [x] Test cases

### Asset Disposal
- [x] Asset Selling
- [x] Scraping Asset
- [x] Test cases

### Asset Splitting and Shift Allocation
- [x] Code cleanup and improvements
- [x] Test cases

### Asset Capitalization
- [x] Code cleanup and improvements
- [x] Test cases

### Misc
- [x] Fix failing test cases (In progress)
- [x] Write additional test cases for better coverage (focus on edge cases)
- [x] Update documentation to reflect changes in the refactored code

### CODE REVIEW

File: erpnext/accounts/doctype/sales_invoice/sales_invoice.py [hunk 1]
```diff
@@ -1518,17 +1563,31 @@ def make_item_gl_entries(self, gl_entries):
...
+				asset,
+				item.base_net_amount,
+				item.finance_book,
+				self.get("doctype"),
+				self.get("name"),
+				self.get("posting_date"),
 			)
 
-		self.check_finance_books(item, asset)
-		return asset
```
Comment (cogk) [inline erpnext/accounts/doctype/sales_invoice/sales_invoice.py:1531]: Hey @khushi8112 :wave:

I see that the call to `check_finance_books` was removed. I'm wondering, is the validation still done somewhere else, or was this intentionally removed?
